include/javabind/collection.hpp

```diff
@@ -330,22 +330,14 @@ namespace javabind
         }
     };
 
-    template <typename T>
-    struct JavaOrderedSetType : JavaSetType<std::set, T>
-    {};
-
-    template <typename T>
-    struct JavaUnorderedSetType : JavaSetType<std::unordered_set, T>
-    {};
-
     /**
      * Converts a native dictionary (e.g. a [map] or [unordered_map]) into a Java Map.
      */
-    template <template<typename...> typename M, typename K, typename V>
-    struct JavaMapType : AssignableJavaType<M<K, V>>
+    template <typename T, template<typename...> typename M>
+    struct JavaMapType : AssignableJavaType<T>
     {
-        using native_type = M<K, V>;
-        using native_boxed_type = M<boxed_t<K>, boxed_t<V>>;
+        using native_type = T;
+        using native_boxed_type = M<boxed_t<typename T::key_type>, boxed_t<typename T::mapped_type>>;
```

Comment:
We seem to be losing template parameters like `Compare` (for ordered), `Hash` and `KeyEqual` (for unordered), and `Allocator`. However, we could pass all of these to `JavaMapType`, couldn't we?

Reply:
Changed it so they are passed here as well